src/measurement/position_measurement.js

```diff
@@ -205,8 +205,10 @@ function measureCharInner(cm, prepared, ch, bias) {
       while (place.coverStart + end < place.coverEnd && isExtendingChar(prepared.line.text.charAt(place.coverStart + end))) ++end
       if (ie && ie_version < 9 && start == 0 && end == place.coverEnd - place.coverStart)
         rect = node.parentNode.getBoundingClientRect()
-      else
-        rect = getUsefulRect(range(node, start, end).getClientRects(), bias)
+        else {
+          if (atomic) {start = 0, end = place.coverEnd - place.coverStart} // cover the entire isolate segment
```

Comment:
Atomic currently restricts cursor motion, but not measuring of characters inside of the span --- and I think it should stay that way.

Reply:
The "atomic" parameter was first called "isolate" (I probably should have left it that way), because I only added it for the new option I'm working on. I renamed it to "atomic" because it was only needed when an isolate segment is also atomic, to prevent the cursor from appearing within the atomc isolate segment, and instead make it appear on the far side of it.